pkg/plugins/manifest.go

```diff
@@ -82,49 +84,133 @@ func readPluginManifest(body []byte) (*pluginManifest, error) {
 	return manifest, nil
 }
 
+// getChksums gets a map of filenames to checksums for all files in a plugin directory.
+//
+// This should be used to validate a plugin manifest.
+func getChksums(dpath, manifestPath string, log log.Logger) (map[string]string, error) {
+	manifestPath = filepath.Clean(manifestPath)
+
+	chksums := map[string]string{}
+	if err := filepath.Walk(dpath, func(path string, fi os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if fi.IsDir() {
+			return nil
+		}
+
+		path = filepath.Clean(path)
+
+		// Handle symbolic links
+		if fi.Mode()&os.ModeSymlink == os.ModeSymlink {
+			finalPath, err := filepath.EvalSymlinks(path)
+			if err != nil {
+				return err
+			}
+
+			log.Debug("Handling symlink", "source", path, "target", finalPath)
+
+			info, err := os.Stat(finalPath)
+			if err != nil {
+				return err
+			}
+			if info.IsDir() {
+				return nil
+			}
+
+			if _, err := filepath.Rel(dpath, finalPath); err != nil {
+				return fmt.Errorf("symbolic link %q targets a file outside of the plugin directory: %q", path, finalPath)
```

Comment:
why do we care?

Reply:
@ryantxu Why should a plugin contain a symlink that points to an external file? At the very least, the current checksumming of plugins won't really work, since we'll be checksumming an external file at the time of packaging, that will very likely point to a file with different content (or not existing at all) when ran by Grafana. I'm also worried it will be an attack vector, so best to avoid the scenario don't you think?